src/provider/controller/IntentHandler.ts

```diff
@@ -47,16 +47,35 @@ export class IntentHandler {
     }
 
     private async raiseWithTarget(intent: IntentWithTarget): Promise<IntentResolution> {
-        const appInfo = await this._directory.getAppByName(intent.target);
+        let appInfo: Application|null;
 
-        if (!appInfo) {
-            throw new FDC3Error(ResolveError.TargetAppNotInDirectory, `No app in directory with name: ${intent.target}`);
-        }
+        const appWindow = this._model.findWindowByAppName(intent.target);
+
+        if (appWindow) {
+            // Target app is running -> ensure that intent is registered in model
+            if (!appWindow.hasIntentListener(intent.type)) {
+                throw new FDC3Error(
+                    ResolveError.TargetAppDoesNotHandleIntent,
+                    `App '${intent.target}' has not registered listeners for intent '${intent.type}'`
+                );
+            }
 
-        if (!(appInfo.intents || []).some(appIntent => appIntent.name === intent.type)) {
-            throw new FDC3Error(ResolveError.TargetAppDoesNotHandleIntent, `App '${intent.target}' does not handle intent '${intent.type}'`);
+            appInfo = appWindow.appInfo;
+        } else {
+            // Target app not running -> Try to find in directory
+            appInfo = await this._directory.getAppByName(intent.target);
+            if (!appInfo) {
+                // TODO: Should this message also mention 'nor running' to cover ad-hoc non-directory apps? [SERVICE-479]
```

Comment:
Resolve before merging

Reply:
@pjbroadbent thoughts?

Reply:
Let's discuss, come up with some wording that makes sense.